pkg/controllers/reconciler.go

```diff
@@ -167,12 +168,27 @@ func (r *reconciler) createObjects(ctx context.Context,
 	return result, nil
 }
 
-func UpgradeImages(image *string, initimage *string) (upgrade bool) {
-	for _, s := range []*string{image, initimage} {
+func UpgradeImages(image *string, initimage *string, kind string) (upgrade bool) {
+	for i, s := range []*string{image, initimage} {
 		if s == nil {
 			continue
 		}
 
+		var envVarValue string
+
+		if i == 1 {
+			envVarValue = os.Getenv(kind + "_IMAGE")
+		} else {
+			envVarValue = os.Getenv(kind + "_INITIMAGE")
+		}
```

Comment:
On the other hand, if the env Variables follow the image names, it would be easier to check here: `os.Getenv(name + "SHA")` ?

Reply:
fixed!